src/main/antora/modules/security/pages/security.adoc

```diff
@@ -776,8 +776,352 @@ public class RestFormAuthCustomStoreIT extends ITBase {
+
+When multiple identity stores are present, a so-called identity store handler (of type `IdentityStoreHandler`) is consulted. Jakara Security provides a default one as explained in <<Securing an endpoint with Basic authentication and multiple identity stores>>. This default handler can be overridden however to provide custom semantics. We'll use a custom handler to enforce a caller authenticates with both identity stores, and we'll combine the roles returned by both in the final result.
+
+==== Populating the identity store
+
+In order to use the identity store, we need to put some data in a database. This is done in the same as in <<Securing an endpoint with Basic authentication and a Database identity store>>.
+
+[NOTE]
+In the custom identity store defined above and in the database identity store here we both use name "john' and password "secret1".
+
+==== Writing the identity store handler
+
+We'll now write the identity store handler:
+
+[source,java]
+----
+@Alternative
+@Priority(APPLICATION)
+@ApplicationScoped
+public class CustomIdentityStoreHandler implements IdentityStoreHandler {
+    
+    @Inject
+    Instance<IdentityStore> identityStores;
+    
+    @Override
+    public CredentialValidationResult validate(Credential credential) {
+        CredentialValidationResult result = null;
+        Set<String> groups = new HashSet<>();
+        
+        for (IdentityStore identityStore : identityStores) {
+            result = identityStore.validate(credential);
+            if (result.getStatus() == NOT_VALIDATED) {
+                // Identity store probably doesn't handle our credential type
+                continue;
+            }
+            
+            if (result.getStatus() == INVALID) {
+                // Identity store handled our credential type and determined its
+                // invalid. End the loop.
+                return INVALID_RESULT;
+            }
+            
+            groups.addAll(result.getCallerGroups());
+        }
+        
+        return new CredentialValidationResult(
+            result.getCallerPrincipal(), groups);
+    }
+}
+----
+
+Since we're overriding an existing CDI bean (the default `IdentityStoreHandler` provided by Jakarta Security), we have to annotate our custom `IdentityStoreHandler` with `@Alternative` and `@Priority(APPLICATION)` as shown above.
+
+With `@Inject` `Instance<IdentityStore> identityStores` CDI will give us a collection of all identity stores in the application. In the case of this example that will be the store behind `@DatabaseIdentityStoreDefinition` and our `CustomIdentityStore`. We can the iterate over those stores in our code, and offer the credentials (the username and password in this example) to each of them.
```

Comment:
These two paragraphs should be callouts: https://docs.asciidoctor.org/asciidoc/latest/verbatim/callouts/

Reply:
That's nice indeed, thanks. Wonder if references to call outs can also be used mid sentence. Did look like it.